### PR TITLE
beta base pwa and IOS

### PR DESCRIPTION
**Pull request by randyhoulahan**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+RUN cd components/AddToCalendar/src && npm install --production
+RUN cd components/Calender && npm install --production
+RUN cd components/conference-cal && npm install --production
+
+
+
+RUN yarn build:ci
+RUN rm -rf node_modules
+RUN npm install --production
+RUN echo $NODE_ENV
```
Comment (sbilodeau) [inline Dockerfile]: why this line?

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+RUN yarn
+RUN cd components/AddToCalendar/src && npm install --production
+RUN cd components/Calender && npm install --production
+RUN cd components/conference-cal && npm install --production
+
+
+
+RUN yarn build:ci
+RUN rm -rf node_modules
+RUN npm install --production
```
Comment (sbilodeau) [inline Dockerfile]: are you using `yarn` or `npm`.

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+RUN cd /usr/src/app/
+COPY . .
+RUN yarn
+RUN cd components/AddToCalendar/src && npm install --production
+RUN cd components/Calender && npm install --production
+RUN cd components/conference-cal && npm install --production
+
+
+
+RUN yarn build:ci
```
Comment (sbilodeau) [inline Dockerfile]: what is this for?

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,31 @@
+FROM node:10-alpine
+
+RUN apk update && apk upgrade && \
+    apk add --no-cache yarn
+
+RUN mkdir -p /usr/src/app
+WORKDIR /usr/src/app
+
+ENV HOST 0.0.0.0
```
[the remark this answers is not in the document]
Comment (randyhoulahan) [reply Dockerfile]: nuxt needs that

File: Dockerfile (hunk 1 above)
Comment (randyhoulahan) [reply Dockerfile]: both, yarn does not do a nice production install.  It is a giant for some reason.

File: Dockerfile (hunk 2 above)
Comment (randyhoulahan) [reply Dockerfile]: both, yarn does not do a nice production install. It is a giant for some reason.

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+ARG BRANCH
+ENV NODE_ENV $BRANCH
+
+RUN cd /usr/src/app/
+COPY . .
+RUN yarn
+RUN cd components/AddToCalendar/src && npm install --production
+RUN cd components/Calender && npm install --production
+RUN cd components/conference-cal && npm install --production
+
```
Comment (blaisefonseca) [inline Dockerfile]: why is this a seprate install, should it not be part of package.json

File: Dockerfile (hunk 3 above)
Comment (randyhoulahan) [reply Dockerfile]: builds with ci env vars

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (sbilodeau) [reply Dockerfile]: ok

File: components/AddToCalendar/src/App.vue [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    data:function(){
+      return{
+        calEvent:calEvent
+    }},
+    components: {
+      AddToCalendar
+    }
+  }
+
+</script>
```
Comment (blaisefonseca) [inline components/AddToCalendar/src/App.vue]: can we follow consistent naming convention? folder name, file name should at least use camel case, I like - seperation with all lowercase

File: Dockerfile (hunk 5 above)
Comment (randyhoulahan) [reply Dockerfile]: they are stand-alone, packages/components.  There are many ways to deal with these.  Design systems, mono repos, learna, yarn workspaces.  We need to decide.  For now they can sit here.

File: components/AddToCalendar/src/App.vue (hunk 6 above)
Comment (randyhoulahan) [reply components/AddToCalendar/src/App.vue]: Yes, I was not sure which to use when I created that.  Components usually are like class names.

File: Dockerfile (hunk 2 above)
Comment (randyhoulahan) [reply Dockerfile]: build with dev dependency node modules , delete then build with production deps.

File: .eslintrc.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+module.exports = {
```
[the remark this answers is not in the document]
Comment (randyhoulahan) [reply .eslintrc.js]: what effects will that have on babel and the web pack building? I am ok with removing but I do not have time to test if it breaks things.

File: components/AddToCalendar/src/App.vue (hunk 6 above)
Comment (sbilodeau) [reply components/AddToCalendar/src/App.vue]: Kebab case
http://wiki.c2.com/?KebabCase

**Comment by sbilodeau** [comment]
You have to fix CircleCI

File: components/Calender/.gitignore [hunk 8]
```diff
@@ -0,0 +1,21 @@
+.DS_Store
```
Comment (sbilodeau) [inline components/Calender/.gitignore]: it's uselss to have gitignore in a subfolder

File: components/Calender/src/App.vue [hunk 9]
```diff
@@ -0,0 +1,147 @@
+<template>
```
Comment (sbilodeau) [inline components/Calender/src/App.vue]: no caps in finlename

File: components/Calender/.gitignore (hunk 8 above)
Comment (randyhoulahan) [reply components/Calender/.gitignore]: yes unless a subtree or module which it is not

File: components/Calender/src/App.vue (hunk 9 above)
Comment (randyhoulahan) [reply components/Calender/src/App.vue]: Vue cli default.  If we impose consistent rules for the style which I am for, they should be automated.  https://www.npmjs.com/package/file-name-linter.

Comment (randyhoulahan) [reply components/Calender/src/App.vue]: That rule may conflict though https://github.com/scbd/accounts.cbd.int/blob/master/Dockerfile

File: components/AddToCalendar/src/App.vue (hunk 6 above)
Comment (randyhoulahan) [reply components/AddToCalendar/src/App.vue]: @sbilodeau & @blaisefonseca I strongly wish we get consistent with our comments and suggestions.   I think comments with respect to code style is a waste of our time.  Looking for consistencies in naming and such.  Why consistent with names but not indenting or semi-colons for example?  Consistency with each serves the same purpose.  Why are consistent names important but indenting?

Consistency has its benefits and I strongly want it.  It's far too much to remember when our own code bases are inconsistent over the years.  Or remembering a syntax styling rule we decided during a discussion.  Too much mental load for me.  I rather use the energy for something else.  That being said can we stop making suggestions on code consistency or implement a team .eslintrc.js in order to save time?

I have indicated before Components have a specific naming, did you guys not remember the discussion?  :-)  I feel I am repeating my self.  All this time could be resolved with a team .eslintrc.js.  

https://vuejs.org/v2/style-guide/#Base-component-names-strongly-recommended
https://angular.io/guide/styleguide
https://hackernoon.com/structuring-projects-and-naming-components-in-react-1261b6e18d76

I would also request that we stop using directives as components.  Angularjs has components. https://www.linkedin.com/pulse/angularjs-1x-directives-vs-components-sachila-ranawaka scan to para: "No more Dom Manipulation"   - will continue this in slack.

File: components/Calender/.gitignore (hunk 8 above)
Comment (randyhoulahan) [reply components/Calender/.gitignore]: Its related to https://github.com/scbd/cbd-events/pull/1#discussion_r229599431.  How shall I resolve?

File: components/Calender/src/App.vue (hunk 9 above)
Comment (randyhoulahan) [reply components/Calender/src/App.vue]: https://vuejs.org/v2/style-guide/#Base-component-names-strongly-recommended
https://angular.io/guide/styleguide
https://hackernoon.com/structuring-projects-and-naming-components-in-react-1261b6e18d76
Please mark resolved or give direction to change.

**Comment by randyhoulahan** [comment]
do not approve this yet docker file and ci changes are coming based on comments.

File: components/AddToCalendar/src/App.vue (hunk 6 above)
Comment (sbilodeau) [reply components/AddToCalendar/src/App.vue]: we used camelCase and kebab-case since we started on angular I don't understand what is new here?

**Pull request closed without merging**